android/app/src/main/kotlin/org/onebusaway/vehicletracker/ui/permissions/PermissionFlow.kt

```
@@ -0,0 +1,277 @@
+                }
+            }
+            Stage.REQUEST_BATTERY -> batteryLauncher.launch(batteryOptimizationIntent(context))
+            Stage.CHECK_LOCATION_SETTINGS -> {
+                val request = LocationSettingsRequest.Builder().addLocationRequest(locationRequest).build()
+                settingsClient.checkLocationSettings(request)
+                    .addOnSuccessListener { stage = Stage.READY }
+                    .addOnFailureListener { exception ->
+                        if (exception is ResolvableApiException) {
+                            try {
+                                settingsResolutionLauncher.launch(IntentSenderRequest.Builder(exception.resolution).build())
+                            } catch (e: IntentSender.SendIntentException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)